PWX-34634: Upgrade build to golang-1.21

zoxpx opened the pull request:
<!--
  Make sure to have done the following:
  [] Signed off your work as per the DCO.
  [] Add unit-tests
-->

**What this PR does / why we need it**:

Security scanners are complaining the `operator` was build via golang-1.20.x
* as a fix, we're upgrading the operator-build to golang-1.21.4  (latest stable)

**Which issue(s) this PR fixes** (optional)
Closes # PWX-34634

**Special notes for your reviewer**:

Quick test:

```
$ docker run --rm -it -v /tmp:/tmp --entrypoint bash zoxpx/px-operator:latest
bash-4.4$ cp /operator /tmp/
bash-4.4$ exit

$ go version /tmp/operator
/tmp/operator: go1.21.4
```

+ manual test: the new operator was able to deploy a functional PX cluster successfully.

piyush-nimbalkar reviewed (approved):
Can we change the go version in the go.mod file too?

zoxpx commented:
> Can we change the go version in the go.mod file too?

Don't think this matters -- but sure, let me re-run the `go mod tidy` using the new golang version

zoxpx commented:
> > Can we change the go version in the go.mod file too?
> 
> Don't think this matters -- but sure, let me re-run the `go mod tidy` using the new golang version

Interesting.. `go mod tidy` didn't update the `go.mod` file
* anyways, I have fixed it manually (see [1445613](https://github.com/libopenstorage/operator/pull/1336/commits/14456135b79d03288c70e6c4e1180072c598a711)), also rebased the PR

zoxpx commented:
Thanks for the review Piyush -- merging the PR, will cherry-pick to `px-rel-23.12.0` branch
